src/odemis/dataio/tiff.py

```diff
@@ -2722,3 +2744,168 @@ def _convert_thermo_fisher_to_odemis_metadata(metadata: str) -> dict:
         logging.warning(f"Failed to parse ThermoFisher metadata: {model.MD_DET_TYPE} - {e}")
 
     return md
+
+## TESCAN
+
+def _parse_tescan_metadata(metadata: str) -> dict:
+    """
+    Parses TESCAN metadata (ini) to dictionary
+    :param metadata: metadata string as ini (TESCAN format)
+    :return: (dict) metadata as dictionary (TESCAN format)
+    """
+    # NOTE:
+    # The metadata should be formatted as ini, but seems to be corrupted
+    # and therefore not parsed correctly by configparser
+    # Therefore we need to parse the metadata manually
+    # The metadata should contain two sections, MAIN and SEM/FIB.
+    # The SEM/FIB section header allows us to determine the beam type,
+    # but it is corrupt and not parsed correctly by configparser.
+
+    # Therefore we need to parse the metadata manually, 
+    # the valid part starts with AccFrames= and ends with a newline
+    valid_md = metadata[metadata.find("AccFrames="):].split("\n")[:-1]
+
+    # these keys are the start of each beam types metadata
+    BEAM_MD_START_KEYS = ["AcceleratorVoltage", "ChamberPressure"]
+
+    md = {}
+    for line in valid_md:
+        try:
+            k, v = (line.split("="))[-2:]
+
+            # NOTE: the first keys for each beam types metadata 
+            # have corrupt binary chars so we need to remove 
+            # these manually
+            for sk in BEAM_MD_START_KEYS:
+                if sk in k:
+                    k = k[k.find(sk):]
+
+        except Exception as e:
+            logging.warning(f"Error: {line} - {e}")    
+            continue
+
+        md[k.lower()] = v
+    return md
+
+
+def _convert_tescan_to_odemis_metadata(metadata: str) -> dict:
+    """
+    Converts TESCAN metadata to odemis metadata format
+    :param metadata: metadata string as ini (TESCAN format)
+    :return: (dict) metadata as dictionary (odemis format)
+    :raises: (Exception): if the metadata cannot be parsed
+    """
+
+    # NOTE: due to metadata corruption, we can't directly read the beam type
+    # the metadata should be in ini format with two sections, MAIN and SEM/FIB
+    # the SEM/FIB section header allows us to determine the beam type, 
+    # but it is corrupt and not parsed correctly by configparser. 
+    # Therefore we need to parse the metadata manually 
+    # and can't determine the beam type
+    # Once fixed:
+    # 1. Change to configparser to parse the metadata correctly
+    # 2. Correctly read the beam type from section headers
+    # 3. Remove the beam type check below (acq_type)
+
+    # parse ini metadata as dictionary 
+    # NOTE: not correctly ini formatted due to corruption
+    try:
+       tmd = _parse_tescan_metadata(metadata)
+    except Exception as e:
+        raise Exception(f"Failed to parse TESCAN metadata: {e}")
```

Comment:
It would be nice to not raise a generic exception here, so we already have a bit more information on what's going wrong.

Reply:
changed to value error